src/openrct2/ride/Vehicle.cpp

```diff
@@ -6638,163 +6580,209 @@ void Vehicle::UpdateAnimationAnimalFlying()
             {
-                animationState = 0;
-                animation_frame += 1;
-                animation_frame &= 7;
-                Invalidate();
-            }
-            break;
-        case CAR_ENTRY_ANIMATION_HELICARS: // Loc6D63F5
-            animationState += _vehicleVelocityF64E08;
-            targetFrame = (animationState >> 18) & 3;
-            if (animation_frame != targetFrame)
-            {
-                animation_frame = targetFrame;
-                Invalidate();
-            }
-            break;
-        case CAR_ENTRY_ANIMATION_MONORAIL_CYCLES: // Loc6D64B6
-            if (num_peeps != 0)
-            {
-                animationState += _vehicleVelocityF64E08;
-                eax = ((animationState >> 13) & 0xFF) << 2;
-                targetFrame = (eax >> 8) & 0xFF;
-                if (animation_frame != targetFrame)
-                {
-                    animation_frame = targetFrame;
-                    Invalidate();
-                }
+                vehicle.animation_frame = targetSeatRotation;
+                vehicle.Invalidate();
             }
-            break;
-        case CAR_ENTRY_ANIMATION_MULTI_DIM_COASTER: // Loc6D65E1
-            if (seat_rotation != target_seat_rotation)
-            {
-                if (animationState <= 0xCCCC)
-                {
-                    animationState += 0x3333;
-                }
-                else
-                {
-                    animationState = 0;
+        }
+    }
+}
 
-                    if (seat_rotation >= target_seat_rotation)
-                        seat_rotation--;
+/**
+ * Animal Flying animates only on chainlift and in an unusual way. Made by Spacek531
+ */
+static void AnimateAnimalFlying(Vehicle& vehicle, const CarEntry& carEntry)
+{
+    vehicle.UpdateAnimationAnimalFlying();
+    // makes animation play faster with vehicle speed
+    uint8_t targetFrame = abs(_vehicleVelocityF64E08) >> carEntry.AnimationSpeed;
+    vehicle.animationState = std::max(vehicle.animationState - targetFrame, 0u);
+}
 
-                    else
-                        seat_rotation++;
+using AnimateFunction = void (*)(Vehicle& vehicle, const CarEntry& carEntry);
 
-                    animation_frame = (seat_rotation - 4) & 7;
-                    Invalidate();
-                }
-            }
-            break;
-        case CAR_ENTRY_ANIMATION_ANIMAL_FLYING:
-            UpdateAnimationAnimalFlying();
-            // makes animation play faster with vehicle speed
-            targetFrame = abs(_vehicleVelocityF64E08) >> 24;
-            animationState = std::max(animationState - targetFrame, 0u);
-            break;
+constexpr static const AnimateFunction animationFunctions[EnumValue(CarEntryAnimation::Count)]{
+    AnimateNone,          AnimateSimpleVehicle,  AnimateSteamLocomotive,  AnimateSwanBoat,
+    AnimateMonorailCycle, AnimateMultiDimension, AnimateObservationTower, AnimateAnimalFlying,
+};
```

Comment:
Using `static_assert` is a better way to assure the size is filled correctly, otherwise it'll just zero-initialize new entries when `Count` increases.
```suggestion
constexpr static const AnimateFunction animationFunctions[]{
    AnimateNone,          AnimateSimpleVehicle,  AnimateSteamLocomotive,  AnimateSwanBoat,
    AnimateMonorailCycle, AnimateMultiDimension, AnimateObservationTower, AnimateAnimalFlying,
};
static_assert(size::size(animationFunctions) == EnumValue(CarEntryAnimation::Count));
```

The same goes for the `SpriteGroupMultiplier` array defined much earlier in the code.

Reply:
Also use TitleCase for static globals.

Reply:
Sprite Group Multiplier would be more appropriately fixed in https://github.com/OpenRCT2/OpenRCT2/pull/19567